### PR TITLE
PeerDAS: Add KZG verification when sampling

### PR DESCRIPTION
**Pull request by nalepae**
Please read commit by commit.

Before this PR,  `VerifyKZGInclusionProofColumn` and `VerifyDataColumnSidecarKZGProofs` were not evaluated when receiving columns from a peer during sampling.

Previously, only the root and the column index were evaluated. This allowed a peer to submit a fabricated column with a correct root and index, which posed a security risk.

Now, `VerifyKZGInclusionProofColumn` and `VerifyDataColumnSidecarKZGProofs `are evaluated for every sampled column, and the corresponding columns are rejected if these evaluations fail.

### CODE REVIEW

File: beacon-chain/sync/validate_data_column.go [hunk 1]
```diff
@@ -23,88 +23,121 @@ import (
...
 	verified, err := peerdas.VerifyDataColumnSidecarKZGProofs(ds)
 	if err != nil {
 		return pubsub.ValidationReject, err
 	}
+
 	if !verified {
 		return pubsub.ValidationReject, errors.New("failed to verify kzg proof of column")
 	}
+
+	// ?????
```
Comment (nalepae) [inline beacon-chain/sync/validate_data_column.go:140]: @nisdas to what specification rule this portion of code does correspond?

Comment (nalepae) [reply beacon-chain/sync/validate_data_column.go:140]: Actually I was more talking about the following block:

```go
	if err := coreBlocks.VerifyBlockHeaderSignatureUsingCurrentFork(parentState, ds.SignedBlockHeader); err != nil {
		return pubsub.ValidationReject, err
	}
```

Comment (nisdas) [reply beacon-chain/sync/validate_data_column.go:140]: > [REJECT] The proposer signature of sidecar.signed_block_header, is valid with respect to the block_header.proposer_index pubkey.

File: beacon-chain/sync/validate_data_column.go [hunk 2]
```diff
@@ -113,35 +146,38 @@ func (s *Service) validateDataColumn(ctx context.Context, pid peer.ID, msg *pubs
...
+
 	if ds.SignedBlockHeader.Header.ProposerIndex != idx {
 		return pubsub.ValidationReject, errors.New("incorrect proposer index")
 	}
 
-	startTime, err := slots.ToTime(uint64(s.cfg.chain.GenesisTime().Unix()), ds.SignedBlockHeader.Header.Slot)
-	if err != nil {
-		return pubsub.ValidationIgnore, err
+	// Add specific debug log.
+	if logrus.GetLevel() >= logrus.DebugLevel {
```
[the remark this answers is not in the document]
Comment (nalepae) [reply beacon-chain/sync/validate_data_column.go]: Fixed in b4ad9143819ddc18f7bc938102e22c7f39c9f2bf.

File: beacon-chain/sync/validate_data_column.go (hunk 1 above)
Comment (nalepae) [reply beacon-chain/sync/validate_data_column.go]: Fixed in 6431659ef2897c05050697b7112901dce836e965

**Review by nisdas — APPROVED** [review]

**Pull request merged**